lib/pure/math.nim

```diff
@@ -141,6 +141,22 @@ when not defined(JS): # C
   proc ln*(x: float32): float32 {.importc: "logf", header: "<math.h>".}
   proc ln*(x: float64): float64 {.importc: "log", header: "<math.h>".}
     ## Computes the natural log of `x`
+else: # JS
+  proc sqrt*(x: float32): float32 {.importc: "Math.sqrt", nodecl.}
+  proc sqrt*(x: float64): float64 {.importc: "Math.sqrt", nodecl.}
+
+  proc ln*(x: float32): float32 {.importc: "Math.log", nodecl.}
+  proc ln*(x: float64): float64 {.importc: "Math.log", nodecl.}
+
+proc log*[T: SomeFloat](x, base: T): T =
+  ## Computes the logarithm ``base`` of ``x``
+  ln(x) / ln(base)
+proc log*(x: float64, base: float32): float64 =
```

Comment:
Don't add this overload, we don't support mixing floating point types everywhere because we know math and what "combinatorial explosion" means.

Reply:
> because we know math and what "combinatorial explosion" means.

Do we? Sorry but I don't. I do know the pain of getting a type mismatch error for similar cases though: https://github.com/nim-lang/Nim/issues/7920